suite-native/firmware/src/screens/FirmwareUpdateInProgressScreen.tsx

```diff
@@ -0,0 +1,191 @@
+import { useFirmware } from '../hooks/useFirmware';
+type NavigationProp = StackNavigationProps<
+    DeviceSettingsStackParamList,
+    DeviceStackRoutes.FirmwareUpdateInProgress
+>;
+
+const bottomButtonsContainerStyle = prepareNativeStyle<{ bottom: number }>((utils, { bottom }) => ({
+    position: 'absolute',
+    left: utils.spacings.sp16,
+    right: utils.spacings.sp16,
+    bottom,
+}));
+
+export const FirmwareUpdateInProgressScreen = () => {
+    const dispatch = useDispatch();
+    const { applyStyle } = useNativeStyles();
+    const navigation = useNavigation<NavigationProp>();
+    const { bottom: bottomSafeAreaInset } = useSafeAreaInsets();
+    const {
+        operation,
+        setIsFirmwareInstallationRunning,
+        confirmOnDevice,
+        firmwareUpdate,
+        progress,
+        status,
+        resetReducer,
+        translatedText,
+        mayBeStucked,
+    } = useFirmware({});
+
+    useEffect(() => {
+        // This will prevent device from being forgotten after firmware update, so discovery will not run again
+        dispatch(setDeviceForceRememberedThunk({ forceRemember: true }));
+
+        return () => {
+            dispatch(setDeviceForceRememberedThunk({ forceRemember: false }));
+            resetReducer();
+        };
+    }, [dispatch, resetReducer]);
+
+    const handleFirmwareUpdateFinished = useCallback(() => {
+        requestPrioritizedDeviceAccess({
+            deviceCallback: () => dispatch(authorizeDeviceThunk()),
+        });
+        navigation.goBack();
+    }, [dispatch, navigation]);
+
+    const startFirmwareUpdate = useCallback(async () => {
+        setIsFirmwareInstallationRunning(true);
+
+        const result = await firmwareUpdate();
+
+        if (!result) {
+            // some error happened probably, handled in redux, we don't want to navigate anywhere
+            return;
+        }
+        if (!result.success) {
+            if (
+                // Action cancelled on device
+                result.payload?.code === 'Failure_ActionCancelled'
+            ) {
+                navigation.navigate(DeviceStackRoutes.FirmwareUpdate);
+            }
+
+            return;
+        }
+        setTimeout(() => {
+            // setting this to false will trigger standart device connection flow
+            setIsFirmwareInstallationRunning(false);
+            handleFirmwareUpdateFinished();
+        }, 5000);
+    }, [
+        setIsFirmwareInstallationRunning,
+        navigation,
+        handleFirmwareUpdateFinished,
+        firmwareUpdate,
+    ]);
+
+    const handleRetry = useCallback(async () => {
+        await TrezorConnect.cancel();
+        resetReducer();
+        startFirmwareUpdate();
+    }, [startFirmwareUpdate, resetReducer]);
+
+    const handleMayBeStucked = useCallback(() => {
+        // todo
+    }, []);
+
+    const handleContactSupport = useCallback(() => {
+        Linking.openURL('https://trezor.io/support');
```

Comment:
We already have `SUITE_LITE_SUPPORT_URL` from `@suite-native/link` on different places.

Reply:
Definitely

Reply:
fixed [61b3b75](https://github.com/trezor/trezor-suite/pull/15184/commits/61b3b75ec0d7f238f34d1faada1ec2054303a0a6)